shap/explainers/_kernel.py

```diff
@@ -17,6 +17,12 @@
 from sklearn.preprocessing import StandardScaler
 from tqdm.auto import tqdm
 from ._explainer import Explainer
+from .._explanation import Explanation
+import time
+
+# Suppress the sklearn warning
+# 'UserWarning: X does not have valid feature names, but <model> was fitted with feature names'
+warnings.filterwarnings('ignore', module='sklearn')
```

Comment:
Seems like a strange thing to filter, just curious where this is coming from? and whether it is possible to fix rather than just ignoring it?

Reply:
Yes, it is a hack. The error is printed repeatedly as the code iterates over the rows of data, so avoiding it is preferable, but not essential. The issues arises if the model is trained on a Pandas dataframe. The shap code only takes the values (line 128: `X = X.values`), so sklearn throws this warning. See [this SO question](https://stackoverflow.com/questions/69326639/sklearn-warning-valid-feature-names-in-version-1-0). I looked into trying to fix the error, rather than just ignoring it, but a lot of methods for Numpy arrays are used and unweaving all of that would require considerable effort.

Reply:
Thank you for pushing on this. I think I have found a much better solution, namely by setting the attribute `feature_names_in_` to `None` (which if non-empty causes sklearn to raise the warning). I make a copy so not to modify the original model.